commons-mongo/src/main/scala/com/avsystem/commons/mongo/BsonReaderInput.scala

```diff
@@ -47,30 +55,12 @@ class BsonReaderIterator[T](br: BsonReader, endCallback: BsonReader => Unit, rea
   }
 }
 
-class BsonReaderListInput private(br: BsonReader) extends ListInput {
-  private val it = new BsonReaderIterator(br, _.readEndArray(), new BsonReaderInput(_))
-
+final class BsonReaderListInput(private val it: BsonReaderIterator[BsonReaderInput]) extends AnyVal with ListInput {
```

Comment:
I do know that, but since I was inlining those single-use helpers, I thought I might as well make it a value class. You never know what they support in Scala 3.14159.

Reply:
What I'm afraid of is that it may actually *hurt* performance because now these classes may get constantly boxed and unboxed all the time:

```scala
val inst = new PointlessValueClass("whatever")
inst.thisMethodWillAlwaysBox()
inst.thisMethodWillAlwaysBox()
```
       0: aload_0
       1: invokespecial #41                 // Method java/lang/Object."<init>":()V
       4: aload_0
       5: putstatic     #43                 // Field MODULE$:LPointlessValueClass$;
       8: aload_0
       9: ldc           #45                 // String whatever
      11: putfield      #17                 // Field inst:Ljava/lang/String;
      14: new           #32                 // class PointlessValueClass
      17: dup
      18: aload_0
      19: invokevirtual #47                 // Method inst:()Ljava/lang/String;
      22: invokespecial #50                 // Method PointlessValueClass."<init>":(Ljava/lang/String;)V
      25: invokevirtual #53                 // Method PointlessValueClass.thisMethodWillAlwaysBox:()V
      28: new           #32                 // class PointlessValueClass
      31: dup
      32: aload_0
      33: invokevirtual #47                 // Method inst:()Ljava/lang/String;
      36: invokespecial #50                 // Method PointlessValueClass."<init>":(Ljava/lang/String;)V
      39: invokevirtual #53                 // Method PointlessValueClass.thisMethodWillAlwaysBox:()V
      42: return
```